incubator/druid/templates/middleManager/statefulset.yaml

```diff
@@ -97,7 +99,7 @@ spec:
         - containerPort: {{ .Values.middleManager.port }}
           name: http
         volumeMounts:
-        - mountPath: /var/druid/
+        - mountPath: /opt/druid/var/druid/
```

Comment:
What's that ?

Reply:
For my understanding :
1.  many default configs use `var/druid` as working data folder.
2. So that `/opt/apache-druid-0.18.1/var/druid` is the default working data folder of druid process in the `druid` image.
3. And `/opt/druid/` is a soft link to `/opt/apache-druid-0.18.1/`
```bash
/opt/apache-druid-0.18.1 # ls -la /opt/
total 0
drwxr-xr-x    1 druid    druid           41 May 13 17:29 .
drwxr-xr-x    1 root     root           118 Jun 30 07:27 ..
drwxr-xr-x    1 druid    druid           25 May 13 17:30 apache-druid-0.18.1
lrwxrwxrwx    1 druid    druid           24 May 13 17:29 druid -> /opt/apache-druid-0.18.1
```
4. So that`/opt/druid/var/druid/` is the default working data folder.

If we change `mountPath` from `/var/druid/` to `/opt/druid/var/druid/`, we don't set config with `/var/druid/` specially any more.

Reply:
this is merged fix from another blocked PR #23069

Reply:
@AWaterColorPen @maver1ck by changing the mount point.. files are actually written to the mounted storage instead of the local FS..

Reply:
It is a good fix. I agree with this PR.